File: 28-find-the-index-of-the-first-occurrence-in-a-string/main.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        
        if needle not in haystack or len(haystack) < len(needle):
            return -1
        
        n = len(haystack)
        for i in range(n):
            if haystack[i: i + len(needle)] == needle:
                return i

        return -1

File: 28-find-the-index-of-the-first-occurrence-in-a-string/main.go
func strStr(haystack string, needle string) int {
    
    if len(haystack) < len(needle) {
        return -1
    }

    for i := 0; i <= len(haystack) - len(needle); i++ {
        if haystack[i:i + len(needle)] == needle {
            return i
        }
    }

    return -1
}
